ngraph/core/src/function.cpp

```diff
@@ -347,6 +356,38 @@ bool Function::visit_attributes(AttributeVisitor& visitor)
     return true;
 }
 
+void Function::add_sinks(const SinkVector& sinks)
+{
+    m_sinks.insert(m_sinks.begin(), sinks.begin(), sinks.end());
+}
+
+void Function::delete_sink(const std::shared_ptr<op::Sink>& sink)
+{
+    string remove_name = sink->get_name();
+    m_sinks.erase(std::remove_if(m_sinks.begin(), 
+                                 m_sinks.end(),
+                                 [remove_name](std::shared_ptr<op::Sink>& s)
+                                 {
+                                     return s->get_name() == remove_name; 
+                                 }), m_sinks.end());
+}
+
+void Function::add_results(const ResultVector& results)
+{
+    m_results.insert(m_results.begin(), results.begin(), results.end());
```

Comment:
We also need at least to call validate and infer.

Reply:
agree, will fix it